e3db/client.py

```diff
@@ -482,51 +482,102 @@ def register(self, registration_token, client_name, public_key, private_key=None
-        if Crypto.get_mode() == 'nist':
-            payload = {
-                'token': registration_token,
-                'client': {
-                    'name': client_name,
-                    'public_key': {
-                        'p384': public_key
+        if public_signing_key == None:
+            if Crypto.get_mode() == 'nist':
+                payload = {
+                    'token': registration_token,
+                    'client': {
+                        'name': client_name,
+                        'public_key': {
+                            'p384': public_key
+                        }
+                    }
+                }
+            else:
+                payload = {
+                    'token': registration_token,
+                    'client': {
+                        'name': client_name,
+                        'public_key': {
+                            'curve25519': public_key
+                        }
                     }
                 }
-            }
         else:
-            payload = {
-                'token': registration_token,
-                'client': {
-                    'name': client_name,
-                    'public_key': {
-                        'curve25519': public_key
+            if Crypto.get_mode() == 'nist':
+                payload = {
+                    'token': registration_token,
+                    'client': {
+                        'name': client_name,
+                        'public_key': {
+                            'p384': public_key
+                        },
+                        'signing_key': {
+                            'ed25519': public_signing_key
+                        }
+                    }
+                }
+            else:
+                payload = {
+                    'token': registration_token,
+                    'client': {
+                        'name': client_name,
+                        'public_key': {
+                            'curve25519': public_key
+                        },
+                        'signing_key': {
+                            'ed25519': public_signing_key
+                        }                    
                     }
                 }
-            }
         response = requests.post(url=url, json=payload)
         self.__response_check(response)
         client_info = response.json()
         backup_client_id = response.headers['x-backup-client']
-
         if backup:
             if private_key is None:
                 raise RuntimeError("Cannot back up client credentials without a private key!")
-
-            config = Config(
-                client_info['client_id'],
-                client_info['api_key_id'],
-                client_info['api_secret'],
-                public_key,
-                private_key,
-                api_url=api_url,
-            )
+            if public_signing_key is not None:
```

Comment:
This if is now done twice in a row, It may make sense to move the private_signing_key check into the below else block

Reply:
Gotcha, I think it will make sense if I also move the `if private_key is None` into the `if public_signing_key is None` block

Reply:
Ah, actually that means I will I have to check `private_key` twice